src/component.js

```diff
@@ -73,7 +73,21 @@ Component.prototype.forceUpdate = function(callback) {
 		let mounts = [];
 		dom = diff(parentDom, vnode, vnode, this._context, parentDom.ownerSVGElement!==undefined, null, mounts, this._ancestorComponent, force, dom);
 		if (dom!=null && dom.parentNode!==parentDom) {
-			parentDom.appendChild(dom);
+			// The component may be rendered somewhere in the middle of the parent's
+			// children. We need to find the nearest DOM sibling to insert our
+			// newly rendered node into.
+			let nextDom;
+			let sibling = this._siblingVNode;
+			while (sibling && !(nextDom = sibling._dom)) {
+				sibling = sibling._component && sibling._component._siblingVNode;
```

Comment:
This code assumes that the next sibling can be found in a sibling `vnode`. Need to check how it behaves for nested trees with a higher depth-level.

Reply:
Don't really think that's a bad assumption (can't see the flaw in it)